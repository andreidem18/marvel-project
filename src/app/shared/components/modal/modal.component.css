.modal-container {
    position: fixed;
    visibility: hidden;
    opacity: 0;
    transition: visibility .2s, opacity .2s linear;
    z-index: 9;
}

.modal-container.active {
    visibility: visible;
    opacity: 1;
    transition: visibility .2s, opacity .2s linear;
}

.modal-container .overlay {
    position: absolute;
    background: rgba(0,0,0,.3);
}


.modal-container .modal-body {
    position: relative;
    z-index: 9;
    background: var(--surface-light);
}
