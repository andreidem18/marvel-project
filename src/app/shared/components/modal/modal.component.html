<div 
    class="modal-container wrap center"
    [ngClass]="{'active': isOpen}"
>
    <div class="modal-body">

        <ng-content ></ng-content>
        
    </div>
    <div 
        class="overlay wrap"
        (click)="close()"
    ></div>
</div>
