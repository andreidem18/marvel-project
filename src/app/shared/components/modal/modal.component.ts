import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'shared-modal-wrapper',
  templateUrl: './modal.component.html',
  styleUrls: ['./modal.component.css']
})
export class ModalComponent {

  @Input()
  public isOpen: boolean = true;

  @Output()
  public isOpenChange = new EventEmitter<boolean>();

  close() {
    this.isOpen = false;
    this.isOpenChange.emit(false);
  }

}
