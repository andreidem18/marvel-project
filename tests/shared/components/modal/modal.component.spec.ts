import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ModalComponent } from '../../../../src/app/shared/components/modal/modal.component';

describe('ModalComponent', () => {
  let component: ModalComponent;
  let fixture: ComponentFixture<ModalComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ModalComponent]
    });
    fixture = TestBed.createComponent(ModalComponent);
    component = fixture.componentInstance;
    compiled = fixture.nativeElement;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match with the snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  it('should be active only if isOpen=true', () => {
    let modalTag = compiled.querySelector('.modal-container');
    expect(modalTag?.classList.contains('active')).toBeTruthy();

    component.isOpen = false;
    fixture.detectChanges();

    modalTag = compiled.querySelector('.modal-container');
    expect(modalTag?.classList.contains('active')).toBeFalsy();
  });

  it('should close when clicking the overlay', () => {
    const overlayTag = compiled.querySelector('.overlay');
    overlayTag?.dispatchEvent(new Event('click'));
    fixture.detectChanges();
    expect(component.isOpen).toBeFalsy();
  });
});
