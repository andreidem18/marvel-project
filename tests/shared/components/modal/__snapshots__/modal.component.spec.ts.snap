// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent should match with the snapshot 1`] = `"<div class="modal-container wrap center active" ng-reflect-ng-class="[object Object]"><div class="modal-body"></div><div class="overlay wrap"></div></div>"`;
